### PR TITLE
Aircraft pathfinding fix

### LINKED ISSUE #1327: Aircraft DLC: Aircrafts take off from taxiways?
### Describe the problem
User reports that when TM:PE is enabled Aircrafts from DLC sometimes take off from taxiway instead of runways

### Steps to reproduce

Not clear what is required to trigger that bug.

### Clip showing the problem with take off
(14:15min and later)
[Watch the video](https://youtu.be/bXAdGNBQVBo?t=870)

### LINKED ISSUE #1329: Aircraft DLC: Aircrafts sometimes stuck at taxiway junction
### Describe the problem
User reports that TM:PE sometimes cause vehicles to stuck at taxiway junctions. Removing TM:PE fixes the problem.

### Steps to reproduce

Not clear how to setup taxiways unfortunately, I'm waiting for savegame to start debugging the problem.

### Screenshots?
image

### PR DESCRIPTION
**Pull request by krzychu124**
TM:PE is searching for path for all AI's in game, so added `vehicleType` converter to handle all explicitly used vehicles by pathfinding. Previously they set vehicle type None which caused issues in path-finding code testing lanes with allowed vehicle types. This change may fix other rare issues that happened in the past for AIs not explicitly managed by TM:PE

closes #1327 
closes #1329 

[Build zip](https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=bugfix/1326-aircraft-pathfinding)

### CODE REVIEW

File: TLM/TLM/Manager/Impl/ExtVehicleManager.cs [hunk 1]
```diff
@@ -1041,6 +1041,54 @@ public class ExtVehicleManager
...
+        /// <summary>
+        /// Converts game VehicleInfo.VehicleType to closest TMPE.API.Traffic.Enums.ExtVehicleType
+        /// </summary>
+        /// <param name="vehicleType"></param>
+        /// <returns></returns>
+        public static ExtVehicleType ConvertToExtVehicleType(VehicleInfo.VehicleType vehicleType) {
+            ExtVehicleType extVehicleType = ExtVehicleType.None;
+            if ((vehicleType & VehicleInfo.VehicleType.Car) != VehicleInfo.VehicleType.None) {
+                extVehicleType = ExtVehicleType.Bus;
+            }
```
Comment (aubergine10) [inline TLM/TLM/Manager/Impl/ExtVehicleManager.cs:1053]: This will always freak me out.

**Comment by aubergine10** [comment]
Should we add a pathfinder version number (similar to how we add version in `WhatsNew.cs`) and if user loads a city with earlier (or no) pathfinder version clear all traffic to ensure all vehicles are using updated paths? #485

File: TLM/TLM/Manager/Impl/ExtVehicleManager.cs [hunk 2]
```diff
@@ -1041,6 +1041,54 @@ public class ExtVehicleManager
             }
         }
 
+        /// <summary>
+        /// Converts game VehicleInfo.VehicleType to closest TMPE.API.Traffic.Enums.ExtVehicleType
+        /// </summary>
+        /// <param name="vehicleType"></param>
+        /// <returns></returns>
+        public static ExtVehicleType ConvertToExtVehicleType(VehicleInfo.VehicleType vehicleType) {
```
Comment (kvakvs) [inline TLM/TLM/Manager/Impl/ExtVehicleManager.cs:1049]: should be a train of if-elseif, not plain if if if? It checks entire set of ifs every time. If the if priority matters, then invert the order of ifs and connect them with else-if chain?

File: TLM/TLM/Manager/Impl/ExtVehicleManager.cs (hunk 1 above)
Comment (aubergine10) [reply TLM/TLM/Manager/Impl/ExtVehicleManager.cs:1053]: If we treat cars as buses, and buses are set to ignore lane arrows, does that mean cars will also ignore lane arrows? Also, why not set to `ExtVehicleType.PassengerCar`?

**Comment by krzychu124** [comment]
@aubergine10 no need here, I think. This applies only to aircrafts taking off so it will start working for new paths. Clearing all vehicles is a bit overkill in this case. Would be cool to add section in the options where user could select which vehicles are cleared.

**Comment by aubergine10** [comment]
> ... Clearing all vehicles is a bit overkill in this case. Would be cool to add section in the options where user could select which vehicles are cleared.

Would it be worth exposing `ExtVehicleManager.DetermineVehicleTypeFromAIType()` as public to make that easier? That way we can generate checkboxen from `ExtVehicleType` enum, also have a method that accepts an `ExtVehicleType` and despawns matching vehicles?

File: TLM/TLM/Manager/Impl/ExtVehicleManager.cs (hunk 1 above)
Comment (krzychu124) [reply TLM/TLM/Manager/Impl/ExtVehicleManager.cs:1053]: This code is for all **unsupported** vehicle types and it's not perfect. There is no vehicle reference so no way to check AI

**Comment by aubergine10** [comment]
Tested this in-game, although without an airport... so I was focussed on all the other vehicles which seem to be working fine :)

- Public transport incl. trains, buses, ferries
- Cargo - trains, ships
- Emergency helicopters
- City services and road-based emergency services
- Cars, etc.

**Review by aubergine10 — APPROVED** [review]
Done another in-game test with a small airport and it seems to be working as advertised.

**Comment by aubergine10** [comment]
@krzychu124 If you can add a pathfinder version check, and want to despawn planes with this new update (assuming #1341 gets merged first):

```csharp
Singleton<SimulationManager>.instance.AddAction(() => {
    UtilityManager.Instance.DespawnVehicles(ExtVehicleType.Plane);
});
```

**Comment by krzychu124** [comment]
> @krzychu124 If you can add a pathfinder version check, and want to despawn planes with this new update (assuming #1341 gets merged first):
> 
> ```cs
> Singleton<SimulationManager>.instance.AddAction(() => {
>     UtilityManager.Instance.DespawnVehicles(ExtVehicleType.Plane);
> });
> ```

I'll implement suggestion from kvakvs and merge, then you can add such thing in your PR 😉

**Comment by aubergine10** [comment]
Will prolly do separate PR for that once both #1341 and #1338 are merged.

**Pull request merged**